core/src/main/java/org/jdbi/v3/core/qualifier/Qualifiers.java

```diff
@@ -36,40 +36,54 @@
 @Beta
 public class Qualifiers implements JdbiConfig<Qualifiers> {
     private static final JdbiCache<AnnotatedElement[], Set<Annotation>> QUALIFIER_CACHE = JdbiCaches.declare(
-            elements -> elements.length == 1 ? elements[0] : new HashSet<>(Arrays.asList(elements)),
-            (Function<AnnotatedElement[], Set<Annotation>>) Qualifiers::getQualifiers);
+        elements -> elements.length == 1 ? elements[0] : new HashSet<>(Arrays.asList(elements)),
+        (registry, elements) -> registry.get(Qualifiers.class).resolveQualifiers(elements));
+    private final Set<Function<AnnotatedElement, Set<Annotation>>> resolvers = new CopyOnWriteArraySet<>();
     private ConfigRegistry registry;
 
-    public Qualifiers() {}
+    public Qualifiers() {
+        resolvers.add(Qualifiers::getQualifierAnnotations);
+    }
 
-    @Override
-    public void setRegistry(ConfigRegistry registry) {
-        this.registry = registry;
+    public Qualifiers(Qualifiers other) {
+        this.resolvers.addAll(other.resolvers);
+        this.registry = null;
+    }
+
+    public Qualifiers addResolver(Function<AnnotatedElement, Set<Annotation>> resolver) {
+        resolvers.add(resolver);
+        QUALIFIER_CACHE.clear(registry);
```

Comment:
you guys seemed pretty wary of doing this but I'm not seeing an issue tbh

Reply:
This is standard procedure in our existing registries. :+1:

Reply:
`clear` did not exist on jdbicache until now though, how can it be SOP?